odoo/addons/base/ir/ir_model.py

```diff
@@ -1208,6 +1208,7 @@ def _module_data_uninstall(self, modules_to_remove):
         datas = self.search([('module', 'in', modules_to_remove)])
         wkf_todo = []
         to_unlink = tools.OrderedSet()
+        undeletable = []
```

Comment:
This list of `(model, res_id)` looks a bit inefficient to me. It also makes the `filter` below a bit tricky.
Why don't you make a recordset, like this?
```python
undeletable = self.browse()
```